packages/tc-markdown-parser/lib/markdown-parser.js

```diff
@@ -0,0 +1,97 @@
+const remarkParse = require('remark-parse');
+const createStream = require('unified-stream');
+const unified = require('unified');
+const createTreecreeperTitleNode = require('./tree-mutators/create-treecreeper-title-node');
+const createTreecreeperDescriptionNode = require('./tree-mutators/create-treecreeper-description-node');
+const createTreecreeperPropertyNodes = require('./tree-mutators/create-treecreeper-property-nodes');
+const setTreecreeperPropertyNames = require('./tree-mutators/set-treecreeper-property-names');
+const coerceTreecreeperPropertiesToType = require('./tree-mutators/coerce-treecreeper-properties-to-type');
+const validateTreecreeperProperties = require('./tree-mutators/validate-treecreeper-properties');
+const stringifyBoast = require('./unist-stringifiers/stringify-boast');
+
+/* @param schema: Treecreeper schema singleton */
+const unifiedProcessor = function(
+	schema,
+	{
+		type,
+		titleNodeName = 'name',
+		descriptionNodeName = 'description',
+		blacklistPropertyNames = [],
+	},
+) {
+	return async () => {
+		await schema.ready();
+
+		const types = schema.getTypes();
+		const { properties } = types.find(({ name }) => name === type);
+		const typeNames = new Set(types.map(({ name }) => name));
+
+		const validateProperty = (key, value) => {
+			return schema.validators.validateProperty(type, key, value);
+		};
+
+		return unified()
+			.use(remarkParse)
+			.use(createTreecreeperTitleNode, {
+				titleNodeName,
+			})
+			.use(createTreecreeperPropertyNodes)
+			.use(createTreecreeperDescriptionNode, {
+				descriptionNodeName,
+			})
+			.use(setTreecreeperPropertyNames, {
+				properties,
+			})
+			.use(coerceTreecreeperPropertiesToType, {
+				properties,
+				typeNames,
+				primitiveTypesMap: schema.primitiveTypesMap,
+				enums: schema.getEnums(),
+			})
+			.use(validateTreecreeperProperties, {
+				validateProperty,
+				propertyNameBlacklist: blacklistPropertyNames,
+			})
+			.use(stringifyBoast, {
+				titleNodeName,
+				descriptionNodeName,
+			});
+	};
+};
+
+const getParser = (
+	schema,
```

Comment:
don't require schema as parameter, or pass around as a variable. Just require tc-schema-sdk at the top of this file and others that need it